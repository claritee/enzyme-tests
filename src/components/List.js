import React from 'react';
import PropTypes from 'prop-types';

import { render } from 'react-dom'

import ListItem from './ListItem';

window.React = React //https://stackoverflow.com/questions/32070303/uncaught-referenceerror-react-is-not-defined?rq=1

/*
 Render a list of items
 @param {Object} props - List of items
 */
function List(props) {
  const { items } = props;
  if (!items.length) {
    // No Items on the list, render an empty message
    return <span className="empty-message">No items in list</span>;
  }

  return (
    <ul className="list-items">
      {items.map(item => <ListItem key={item} item={item} />)}
    </ul>
  );
}

List.propTypes = {
  items: PropTypes.array,
};

List.defaultProps = {
  items: [],
};

export default List;