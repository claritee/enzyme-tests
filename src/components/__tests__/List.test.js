import { render } from '../../enzyme';

import List from '../List';
import { wrap } from 'module';

describe('List tests', () => {
  it('renders list-items', () => {
    const items = ['one', 'two', 'three'];
    const wrapper = render(<List items={items} />);

    wrapper.addClass('foo');

    // Expect the wrapper object to be defined
    // (1) Before ListItem using shallow and (2) using mount and (3) static rendering
    expect(wrapper.find('.list-items')).toBeDefined();
    expect(wrapper.find('.item')).toHaveLength(items.length);

    // After ListItem using shallow
    // expect(wrapper.find('ListItem')).toBeDefined();
    // expect(wrapper.find('ListItem')).toHaveLength(items.length);
  });

  it('renders a list item', () => {
    const items = ['Thor', 'Loki'];
    const wrapper = render(<List items={items} />);

    // Check if an element in the Component exists
    //expect(wrapper.contains(<li key='Thor' className="item">Thor</li >)).toBeTruthy();

    // With static rendering
    expect(wrapper.find('[key=Thor]')).toBeTruthy();
  });

  it('renders correct text in item', () => {
    const items = ['John', 'James', 'Luke'];
    const wrapper = render(<List items={items} />);

    //Before: (1) shallow without ListItem and (2) mount Expect the child of the first item to be an array
    //expect(wrapper.find('.item').get(0).props.children).toEqual('John');

    // After: static rendering 
    expect(wrapper.find('.item').get(0).children[0].data).toEqual('John');
  });
});