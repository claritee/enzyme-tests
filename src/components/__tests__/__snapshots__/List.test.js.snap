// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List tests renders correctly 1`] = `
<ul
  className="list-items"
>
  <li
    className="item"
  >
    A
  </li>
  <li
    className="item"
  >
    B
  </li>
  <li
    className="item"
  >
    C
  </li>
</ul>
`;
